plugins/source/gcp/resources/services/networkconnectivity/internal_ranges.go

```diff
@@ -0,0 +1,50 @@
+package networkconnectivity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"google.golang.org/api/networkconnectivity/v1"
+
+	"github.com/cloudquery/cloudquery/plugins/source/gcp/client"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+)
+
+func InternalRanges() *schema.Table {
+	return &schema.Table{
+		Name:        "gcp_network_connectivity_internal_ranges",
+		Description: `https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/v1/projects.locations.internalRanges/list`,
+		Resolver:    fetchInternalRanges,
+		Multiplex:   client.ProjectMultiplexEnabledServices("networkconnectivity.googleapis.com"),
+		Transform:   client.TransformWithStruct(&networkconnectivity.InternalRange{}, transformers.WithPrimaryKeys("Name")),
+		Columns: []schema.Column{
+			{
+				Name:     "project_id",
+				Type:     arrow.BinaryTypes.String,
+				Resolver: client.ResolveProject,
+			},
+		},
+	}
+}
+
+func fetchInternalRanges(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
+	c := meta.(*client.Client)
+
+	ns, err := networkconnectivity.NewService(ctx)
+	if err != nil {
+		return err
+	}
+
+	parent := fmt.Sprintf("projects/%s/locations/%s", c.ProjectId, "global")
```

Comment:
Aren't there other locations that can be viewed here?
See https://github.com/cloudquery/cloudquery/blob/main/plugins/source/gcp/resources/services/kms/locations.go for example on fetching relevant locations prior fetching the data.

Additionally, if `global` isn't returned, it might be a good idea to make this data available separately (either a separate API call or a separate table).

Reply:
Thanks for the suggestion, Added a new parent table for the network connectivity location.